libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -760,12 +761,18 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 	if (functionCallKind == FunctionCallKind::TypeConversion)
 	{
-		solAssert(
-			_functionCall.expression().annotation().type->category() == Type::Category::TypeType,
-			"Expected category to be TypeType"
-		);
 		solAssert(_functionCall.arguments().size() == 1, "Expected one argument for type conversion");
-		define(_functionCall, *_functionCall.arguments().front());
+		Type const* fromType = _functionCall.arguments().front()->annotation().type;
+		Type const* toType = _functionCall.expression().annotation().type;
+
+		solAssert(toType->category() == Type::Category::TypeType, "Expected category to be TypeType");
+		Type const* toActualType = dynamic_cast<TypeType const*>(toType)->actualType();
+
+		if (fromType->category() == Type::Category::TypeType && toActualType->category() == Type::Category::Address)
+			define(_functionCall) << linkerSymbol(dynamic_cast<TypeType const&>(*fromType)) << "\n";
+		else
+			define(_functionCall, *_functionCall.arguments().front());
```

Comment:
I know this has been like this, but why don't we call the conversion function?

Reply:
Oh it does call the conversion function internally!

Reply:
Yeah. The conversion is there.